cw/cw305/capture.py

```diff
@@ -372,68 +418,66 @@ def capture_aes_fvsr_key_batch(ot, ktp, capture_cfg, scope_type):
             scope.num_segments = min(rem_num_traces, scope.num_segments_max)
             scope.arm()
 
-            # Start batch encryption.
+            # Start batch encryption. In order to increase capture rate, after the first batch
+            # encryption, send "g" to the device to generate random keys and plaintexts at the
+            # device when this script is getting waves from the scope.
+            if is_first_batch:
+                ot.target.simpleserial_write("g", scope.num_segments_actual.to_bytes(4, "little"))
+                is_first_batch = False
             ot.target.simpleserial_write("f", scope.num_segments_actual.to_bytes(4, "little"))
+            next_batch_size = min(rem_num_traces - scope.num_segments_actual,
+                                  scope.num_segments_max)
+            if next_batch_size > 0:
+                ot.target.simpleserial_write("g", next_batch_size.to_bytes(4, "little"))
```

Comment:
Thanks @vogelpi. This worked and I also thought that it would increase the capture rate too since we are using serial communication less this way but it didn't change at all. Maybe we have a bottleneck somewhere else.

Reply:
Thanks for changing it. I think the code is more reliable like this. Don't worry about the capture rate.